pkg/docker/parseimage.go

```diff
@@ -24,25 +24,61 @@ package docker
 import (
 	"fmt"
 	"regexp"
+	"strings"
 )
 
-var prefix = regexp.MustCompile("docker-pullable://")
-var digestRegexp = regexp.MustCompile("@sha256:([a-zA-Z0-9]+)$")
+var dockerPullableRegexp = regexp.MustCompile("^docker-pullable://(.+)@sha256:([a-zA-Z0-9]+)$")
+var dockerRegexp = regexp.MustCompile("^docker://sha256:([a-zA-Z0-9]+)$")
+var imageRegexp = regexp.MustCompile("^(.+)@sha256:([a-zA-Z0-9]+)$")
 
 // ParseImageIDString parses an ImageID that can pull an image from docker
 // Example image id:
 //   docker-pullable://registry.kipp.blackducksoftware.com/blackducksoftware/hub-registration@sha256:cb4983d8399a59bb5ee6e68b6177d878966a8fe41abe18a45c3b1d8809f1d043
 func ParseImageIDString(imageID string) (string, string, error) {
-	str := imageID
-	match := prefix.FindStringIndex(str)
-	if len(match) > 0 && match[0] == 0 {
-		str = str[match[1]:]
+	name, digest, err := ParseDockerPullableImageString(imageID)
+	// Since the GO doesn't support for "don't start with string regex", had an ugly fix with HasPrefix in below line
+	if err == nil || strings.HasPrefix(imageID, "docker-pullable://") {
+		return name, digest, err
```

Comment:
I'm not sure I get this case.  Either the function returns an err and thus fails, or it success and err == nil.  Why do you need to do a HasPrefix check?  If err == nil then it succeed by definition, and if err != nil that name and digest are empty.

Reply:
For the missing image name test case, the dockerPullableRegexp will error out and the err will be != nil and hence the code will start executing the next imageregex. The imageregex will not provide any error to missing image name test case and it will return image name as docker-pullabe:// and sha as <<some value>>. Golang doesn't support "don't start with string regex". Hence we need to add a hack to stop terminating if the image id starts with docker-pullable://